solvebio/exporters.py

```diff
@@ -247,6 +261,98 @@ def write(self, filename):
             f.close()
 
 
+class FlatCSVExporter(CSVExporter):
+    """
+    This class includes helper functions to export
+    a SolveBio Query object to a CSV file without altering the data structure.
+    """
+    name = 'flat-csv'
+
+    def export(self, filename=None, **kwargs):
+        if not filename:
+            raise Exception(
+                'The "filename" parameter is required to export.')
+
+        filename = os.path.expanduser(filename)
+        title = 'Exporting query to: {0}'.format(filename)
+
+        result_count = len(self.query)
+        if result_count <= 0:
+            raise AttributeError('No results found in query!')
+
+        self.rows = []
+        self.fields = set([])
+
+        if self.show_progress:
+            progress_bar = pyprind.ProgPercent(
+                result_count,
+                title=title,
+                track_time=False)
+        else:
+            print(title)
+
+        for record in self.query:
+            self.current_row = {}
+            self.process_record(record)
+            self.rows.append(self.current_row)
+            if self.show_progress:
+                progress_bar.update()
+
+        self.write(filename=filename)
+        print('Export complete!')
+
+    def process_record(self, record, root_key=''):
+        """Process a row of json data
+        """
+        if root_key != '':
+            root_key += '.'
+
+        # Tree traversal
+        for key, value in six.iteritems(record):
+            field = root_key + key
+            # If the value is an object, process it recursively
+            if isinstance(value, dict):
+                self.process_record(value, field)
+            # If the value is a list, parse each item
+            elif isinstance(value, list):
+                for index, item in enumerate(value):
+                    field_name = field + '.' + str(index)
+                    if isinstance(item, dict):
+                        self.process_record(item, field_name)
+                    else:
+                        self.fields.add(field_name)
+                        self.current_row[field_name] = self.make_string(item)
+            # If we reached a leaf, add field and value
+            else:
+                self.fields.add(field)
+                self.current_row[field] = self.make_string(value)
+
+    @staticmethod
+    def make_string(value):
+        try:
+            return str(value)
+        except UnicodeEncodeError:
+            return ''.join(
+                [s.encode('ascii', 'backslashreplace') for s in value]
+            )
+
+    def write(self, filename):
+        if sys.version_info >= (3, 0, 0):
+            f = open(filename, 'w', newline='')
+        else:
+            f = open(filename, 'wb')
+
+        fieldnames = sorted(list(self.fields))
+        try:
+            writer = csv.DictWriter(f, fieldnames=fieldnames)
+            writer.writerow(dict(zip(fieldnames, fieldnames)))
```

Comment:
I read this comment here: https://github.com/solvebio/solvebio-python/blob/28cf4e26a36bae2665dfd48ce0398e448546f8ee/solvebio/exporters.py#L252